### PR TITLE
[WFLY-7680] Lazily import transactions from remote clients

### PR DESCRIPTION
**Pull request by dmlloyd**
https://issues.jboss.org/browse/WFLY-7860

### CODE REVIEW

File: ejb3/src/main/java/org/jboss/as/ejb3/remote/EJBRemoteTransactionPropagatingInterceptor.java [hunk 1]
```diff
@@ -1,92 +0,0 @@
...
-        if (context.hasTransaction()) {
-            // TODO: WFLY-7860
-            try {
-                transactionManager.resume(context.getTransaction());
-            } catch (SystemException e) {
-                try {
-                    EJBComponent component = getComponent(context, EJBComponent.class);
-                    // SystemException + server suspended equals the transaction is new and the request
-                    // for new transaction is being rejected
-                    if (component != null && component.getEjbSuspendHandlerService().isSuspended()) {
```
Comment (fl4via) [inline ejb3/src/main/java/org/jboss/as/ejb3/remote/EJBRemoteTransactionPropagatingInterceptor.java]: @dmlloyd shouldn't this check be moved elsewhere? This is used for graceful shutdown.

Comment (dmlloyd) [reply ejb3/src/main/java/org/jboss/as/ejb3/remote/EJBRemoteTransactionPropagatingInterceptor.java]: You're right; I made a note to do this but then I forgot anyway.  Thanks.

File: ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java [hunk 2]
```diff
@@ -155,6 +155,31 @@ public Object processInvocation(InterceptorContext invocation) throws Exception
...
     }
 
+    protected Object invokeInImportedTx(InterceptorContext invocation, EJBComponent component) throws Exception {
+        Transaction tx;
+        try {
+            tx = invocation.getTransaction();
+        } catch (SystemException ex) {
+            // SystemException + server suspended equals the transaction is new and the request
+            // for new transaction is being rejected
+            if (component != null && component.getEjbSuspendHandlerService().isSuspended()) {
```
Comment (dmlloyd) [inline ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java]: This is the location of the relocated check.

**Pull request merged**